frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1527,18 +1516,29 @@ impl<T: Config> Pallet<T> {
 				err
 			})
 	}
+
+	/// record the weight of the given `supports`.
+	fn weigh_supports(supports: &Supports<T::AccountId>) {
+		let active_voters = supports
+			.iter()
+			.map(|(_, x)| x)
+			.fold(Zero::zero(), |acc, next| acc + next.voters.len() as u32);
+		let desired_targets = supports.len() as u32;
```

Comment:
Beyond the scope of this PR, but could when we create the `Supports` could we put it in a struct that keeps track of the total number of voters with `SupportsWithCount { supports: Supports<_>, voter_count: u32 }` so we don't need this extra iteration? I think this would be most beneficial if targets was unbounded, which is not the case at the moment so I think worst case is still technically constant relative to max target count. (Assuming Vec::len is a constant op)

Reply:
yeah I totally agree, I hate the fact that I had to add this iteration here. Feel free to make an issue in the NPoS project from your suggestion.